packages/relay-compiler/transforms/DeclarativeConnectionMutationTransform.js

```diff
@@ -65,11 +65,18 @@ function visitScalarField(field: ScalarField): ScalarField {
     return field;
   }
   const schema = this.getContext().getSchema();
-  if (!schema.isId(field.type)) {
+
+  if (
+    ['ID', 'ID!', '[ID]', '[ID!]', '[ID!]!'].includes(
```

Comment:
What's your take on this approach? Problematic?

Reply:
you can use `if (!schema.isId(schema.getRawType(field.type)))` to unwrap NonNull/List wrappers and get at the underlying type.